src/main/scala/scynamo/ScynamoEncoder.scala

```diff
@@ -12,119 +12,150 @@ import shapeless.tag.@@
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue
 
 import java.time.Instant
-import java.util.UUID
+import java.util.{Collections, UUID}
+import scala.collection.compat._
+import scala.collection.immutable.Seq
 import scala.concurrent.duration.{Duration, FiniteDuration}
-import scala.jdk.CollectionConverters._
 
 trait ScynamoEncoder[A] { self =>
   def encode(value: A): EitherNec[ScynamoEncodeError, AttributeValue]
-
-  def contramap[B](f: B => A): ScynamoEncoder[B] = value => self.encode(f(value))
+  def contramap[B](f: B => A): ScynamoEncoder[B] =
+    ScynamoEncoder.instance(value => self.encode(f(value)))
 }
 
 object ScynamoEncoder extends DefaultScynamoEncoderInstances {
   def apply[A](implicit instance: ScynamoEncoder[A]): ScynamoEncoder[A] = instance
+
+  // SAM syntax generates anonymous classes because of non-abstract methods like `contramap`.
+  private[scynamo] def instance[A](f: A => EitherNec[ScynamoEncodeError, AttributeValue]): ScynamoEncoder[A] = f(_)
```

Comment:
I assume you wanted to move this, in order to avoid the generation of anonymous classes every time?  I'm not sure what exactly this `instance` method changes... :thinking:

Reply:
The best way to check is to clean compile before the change and after the change - you should see a bunch of anon classes disappear 😬 - it's really a bug in the Scala compiler that can't be fixed without breaking binary compatibility.

Reply:
Well TIL :+1: